ops/_main.py

```diff
@@ -513,6 +513,16 @@ def run(self):
             self._emit(self.charm, self._dispatcher.event_name)
             self._commit(self.charm.framework)
             self._close()
+        except _model.InvalidSchemaError as e:
```

Comment:
Now that the new-charm-for-deferred-events change has landed, this needs to be in `run_deferred` too.